src/NuGet.Core/NuGet.PackageManagement/NuGetPackageManager.cs

```diff
@@ -3279,6 +3327,19 @@ var projectUniqueNamesForBuildIntToUpdate
             await OpenReadmeFile(buildIntegratedProject, nuGetProjectContext, token);
         }
 
+        private static IReadOnlyList<string> GetFilesToBeUpdated(RestoreResultPair result)
+        {
+            List<string> filesToBeUpdated = new(3); // We know that we have 3 files.
+            filesToBeUpdated.Add(result.Result.LockFilePath);
+
+            foreach (MSBuildOutputFile msbuildOutputFile in result.Result.MSBuildOutputFiles)
```

Comment:
Would it be better to just call `AddRange()`?

Reply:
AddRange causes more allocations, because this way we don't need an extra Select

Reply:
Is `result.Result.MSBuildOutputFiles` an `IEnumerable<T>` or something more specific?  If we're worried about allocations, then a `for` loop is faster, as long as we have a count/length property and a get indexer.

Reply:
It's an IEnumerable :(